clang/test/Driver/linker-wrapper-sycl.cpp

```diff
@@ -47,6 +47,11 @@
 // RUN: clang-linker-wrapper -sycl-device-libraries=%t.devicelib.o -sycl-post-link-options="SYCL_POST_LINK_OPTIONS" -llvm-spirv-options="LLVM_SPIRV_OPTIONS" "--host-triple=x86_64-unknown-linux-gnu" "--triple=spir64" "--linker-path=/usr/bin/ld" -shared "--" HOST_LINKER_FLAGS "-dynamic-linker" HOST_DYN_LIB "-o" "a.out" HOST_LIB_PATH HOST_STAT_LIB %t.o --dry-run 2>&1 | FileCheck -check-prefix=CHK-SHARED %s
 // CHK-SHARED: "{{.*}}clang"{{.*}} -fPIC
 
+// RUN: rm %T/linker_wrapper_dump || true
+// RUN: clang-linker-wrapper -sycl-dump-device-code=%T/linker_wrapper_dump -sycl-device-libraries=%t.devicelib.o "--host-triple=x86_64-unknown-linux-gnu" "--triple=spir64" "--linker-path=/usr/bin/ld" -shared "--" HOST_LINKER_FLAGS "-dynamic-linker" HOST_DYN_LIB "-o" "a.out" HOST_LIB_PATH HOST_STAT_LIB %t.o --dry-run
+// RUN: ls %T/linker_wrapper_dump | FileCheck -check-prefix=CHK-SYCL-DUMP-DEVICE %s
+// CHK-SYCL-DUMP-DEVICE: {{.*}}.spv
```

Comment:
This test is wrong in multiple places.
1. `%T/linker_wrapper_dump` is directory, so `rm %T/linker_wrapper_dump` always fails.
2. Why do we need to remove any directories, if the next command is executed in dry-run mode and therefore should not create any directories or files?
3. Lines 52 and 53 are checking that `%T/linker_wrapper_dump` directory has files with `.spv` file extension. These are not supported to be created in dry-run mode. 

I removed the test due to reasons listed above.

Reply:
You are right. I will try to resolve this test in the following PR then.